src/utils/get-transition-duration.ts

```diff
@@ -109,14 +111,43 @@ export const getTransitionDurationFromTheme = (
   return extractDurationFromPreset(tokensArray, props.theme);
 };
 
+// getTransitionDuration is passed to timeout in CSSTransitions which defines the entire timeframe of the transition.
+// Reduced motion is not added here because it has no visual changes and might cause issues in rendering transitions.
 export const getTransitionDuration = (
   defaultPath?: string,
   overridePath?: string | false,
 ) => (props: ThemeProp & {overrides?: unknown}) => {
-  const token = getToken(props, defaultPath, overridePath, 'transitionPreset');
+  const token = getToken(
+    props,
+    defaultPath,
+    overridePath,
+    'transitionPreset',
+  ) as MQ<TransitionToken> | MQ<TransitionToken>[];
 
   if (!token) return {enter: 0, exit: 0, appear: 0} as TimeoutType;
 
+  if (isResponsive(token, props.theme.breakpoints)) {
+    return Object.entries(token).reduce(
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      (acc, [bp, transitionPresetToken]) => {
+        let currentDuration;
+        if (isArrayLikeObject(transitionPresetToken)) {
+          const tokensArray = Object.values(transitionPresetToken) as string[];
+          currentDuration = extractDurationFromPreset(tokensArray, props.theme);
+        } else {
+          const tokensArray = [transitionPresetToken];
+          currentDuration = extractDurationFromPreset(tokensArray, props.theme);
+        }
```

Comment:
This is a bit of personal preference, you can write it like so:
```suggestion
      (acc, [, transitionPresetToken]) => {
        const tokensArray: string[] = isArrayLikeObject(transitionPresetToken)
          ? transitionPresetToken
          : [transitionPresetToken];

        const currentDuration = extractDurationFromPreset(
          tokensArray,
          props.theme,
        );
```

Reply:
This looks neat 🔥. Thanks!